client/network/src/warp_request_handler.rs

```diff
@@ -16,58 +16,57 @@
 
```

Comment:
Even though there's a generic backend for generating warp proofs this is still quite coupled to GRANDPA and feels a bit wrong to add it to the generic layer of networking.

Reply:
I could not think of a good place to put it. Leaving it in a stand-alone crate would introduce a circular dependency because it uses type from `sc-network` and it no really designed for pluggable sync algorithms. Ideally there should be a separate `sc-network-api` crate which only contains the network interface, but not the actual implementation. This however turned out to be a substantial refactoring outside of the scope of this PR.

Reply:
We can make this more abstract in the future, there's no point blocking this PR now on this since the only consensus engine we have right now that supports Warping is GRANDPA/BABE anyway.